onnxruntime/core/util/math_cpu.cc

```diff
@@ -458,12 +463,14 @@ void Gemm<double, CPUMathUtil>(
     double* C,
     CPUMathUtil* /*provider*/,
     MLDataType /*math_type*/) {
-// Since MKLML does not have an implementation for Gemm (double type), check if we can fallback to Eigen 
-#if defined(USE_EIGEN_FOR_BLAS)
-  GemmEigen<double>(TransA, TransB, M, N, K, alpha, A, B, beta, C);
-#else
-  #error "No implementation found for Gemm that supports double type"
-#endif
+    int lda = gsl::narrow_cast<int>((TransA == CblasNoTrans) ? K : M);
```

Comment:
why do you use  gsl::narrow_cast?

Reply:
Hi @snnn - That line is exactly the same as the line in the existing float template specialization of the Gemm template. Kept it as it is in the newly added template specializations.